addons/mail/models/mail_thread.py

```diff
@@ -4141,15 +4141,14 @@ def _get_mail_thread_data(self, request_list):
         if 'attachments' in request_list:
             res['attachments'] = self._get_mail_thread_data_attachments()._attachment_format()
         if 'followers' in request_list:
-            res['followers'] = [{
-                'id': follower.id,
-                'partner_id': follower.partner_id.id,
-                'name': follower.name,
-                'display_name': follower.display_name,
-                'email': follower.email,
-                'is_active': follower.is_active,
-                'partner': follower.partner_id.mail_partner_format()[follower.partner_id],
-            } for follower in self.message_follower_ids]
+            domain = [
+                ("res_id", "=", int(self.id)),
+                ("res_model", "=", self._name),
+            ]
+            self_domain = domain + [['partner_id', '=', self.env.user.partner_id.id]]
+            res['followersCount'] = len(self.message_follower_ids)
+            res['selfFollower'] = self.env['mail.followers'].search(self_domain, limit=1)._follower_format()[0] if self.message_is_follower else False
```

Comment:
You could just make the `message_is_follower` return the record id instead of a boolean and combine it with browse and format.
That way you don't need a search and a domain.

You could also directly return the record and have it formatted. You could make an other computed field that duplicate the logic of is follower but return the record instead. But this is not really worth it I guess?